src/chainstate/stacks/index/storage.rs

```diff
@@ -83,12 +98,34 @@ impl<T: MarfTrieId> BlockMap for TrieFileStorage<T> {
     }
 
     fn get_block_hash_caching(&mut self, id: u32) -> Result<&T, Error> {
-        if !self.data.block_hash_cache.contains_key(&id) {
-            self.data
-                .block_hash_cache
-                .insert(id, self.get_block_hash(id)?);
+        if self.cache.load_block_hash(id).is_none() {
```

Comment:
```suggestion
        if self.is_block_hash_cached(id) {
```

Reply:
I think you mean `if !self.is_block_hash_cached(id) {`, but I take your meaning.